### PR TITLE
Editorial: Create OrdinarySetWithOwnDescriptor

### LINKED ISSUE #719: IDL needs hook for subset of OrdinarySet
For https://github.com/heycam/webidl/issues/201 we need a way to invoke step 3-onward of OrdinarySet.

Would something like OrdinarySetWithOwnDesc(O, P, V, Reciever, ownDesc) be okay? Should this new abstract operation be a subsection or sibling section of OrdinarySet?

### PR DESCRIPTION
**Pull request by TimothyGu**
Closes #719.

/cc @annevk

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -6900,6 +6900,16 @@
         <emu-alg>
           1. Assert: IsPropertyKey(_P_) is *true*.
           1. Let _ownDesc_ be ? _O_.[[GetOwnProperty]](_P_).
+          1. Return OrdinarySetWithOwnDescriptor(_O_, _P_, _V_, _Receiver_, _ownDesc_).
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-ordinarysetwithowndescriptor" aoid="OrdinarySetWithOwnDescriptor">
```
Comment (ljharb) [inline spec.html]: It'd be nice to include a comment here that this is used in WebIDL, but I think that's been shot down before.

Comment (annevk) [reply spec.html]: Yup: https://github.com/tc39/ecma262/pull/398#issuecomment-187724667. (I still think it would be useful...)

**Comment by bterlson** [comment]
I'm kind of sad about all this indirection but alright, alright.

**Comment by annevk** [comment]
@bterlson "Please leave your sense of logic at the door, thanks!"

You're welcome 😛

**Pull request merged**